src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -2981,8 +2986,276 @@ void Schedule::invalidNamePattern( const std::string& namePattern,  std::size_t
+
+
+std::string well_msg(const std::string& well, const std::string& msg) {
+    return "Well: " + well + " " + msg;
+}
+
+
+}
+
+bool Schedule::cmp(const Schedule& sched1, const Schedule& sched2, std::size_t report_step) {
+    int count = 0;
+    for (const auto& wname : sched1.wellNames(report_step)) {
+        const auto& well1 = sched1.getWell(wname, report_step);
+        const auto& well2 = sched2.getWell(wname, report_step);
+        int well_count = 0;
+        {
+            const auto& connections2 = well2.getConnections();
+            const auto& connections1 = well1.getConnections();
+
+            if (not_equal(connections1.size(), connections2.size(), well_msg(well1.name(), "Connections: size")))
+                return false;
+
+            for (std::size_t icon = 0; icon < connections1.size(); icon++) {
+                const auto& conn1 = connections1[icon];
+                const auto& conn2 = connections2[icon];
+                well_count += not_equal( conn1.getI(), conn2.getI(), well_msg(well1.name(), "Connection: I"));
+                well_count += not_equal( conn1.getI() , conn2.getI() , well_msg(well1.name(), "Connection: I"));
+                well_count += not_equal( conn1.getJ() , conn2.getJ() , well_msg(well1.name(), "Connection: J"));
+                well_count += not_equal( conn1.getK() , conn2.getK() , well_msg(well1.name(), "Connection: K"));
+                well_count += not_equal( conn1.state() , conn2.state(), well_msg(well1.name(), "Connection: State"));
+                well_count += not_equal( conn1.dir() , conn2.dir(), well_msg(well1.name(), "Connection: dir"));
+                well_count += not_equal( conn1.complnum() , conn2.complnum(), well_msg(well1.name(), "connection: complnum"));
+                well_count += not_equal( conn1.segment() , conn2.segment(), well_msg(well1.name(), "Connection: segment"));
+                well_count += not_equal( conn1.kind() , conn2.kind(), well_msg(well1.name(), "Connection: CFKind"));
+
+
+                well_count += not_equal( conn1.CF(), conn2.CF(), well_msg(well1.name(), "Connection: CF"));
+                well_count += not_equal( conn1.Kh(), conn2.Kh(), well_msg(well1.name(), "Connection: Kh"));
+                well_count += not_equal( conn1.rw(), conn2.rw(), well_msg(well1.name(), "Connection: rw"));
+                well_count += not_equal( conn1.depth(), conn2.depth(), well_msg(well1.name(), "Connection: depth"));
+
+                well_count += not_equal( conn1.r0(), conn2.r0(), well_msg(well1.name(), "Connection: r0"));
+                well_count += not_equal( conn1.skinFactor(), conn2.skinFactor(), well_msg(well1.name(), "Connection: skinFactor"));
+                well_count += not_equal( conn1.wellPi(), conn2.wellPi(), well_msg(well1.name(), "Connection: PI"));
+
+            }
+        }
+
+        if (well1.isMultiSegment()) {
+            const auto& segments1 = well1.getSegments();
+            const auto& segments2 = well2.getSegments();
```

Comment:
What does `well2.getSegments()` return if `well2` is not a multi-segmented well?

Reply:
`throw std::logic_error()`